frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-internal/details-card.tsx

```diff
@@ -110,14 +165,31 @@ const DetailsCard: React.FC<DashboardItemProps> = ({
             {infrastructurePlatform}
           </DetailItem>
           <DetailItem title={t('ceph-storage-plugin~Mode')}>Internal</DetailItem>
-          <DetailItem
-            key="version"
-            title={t('ceph-storage-plugin~Version')}
-            isLoading={!subscriptionLoaded}
-            error={subscriptionLoaded && !ocsVersion}
-          >
-            {ocsVersion}
-          </DetailItem>
+          {!isEmpty(filteredVersions) && ocsSubscription?.spec?.installPlanApproval === 'Manual' ? (
+            <DetailItem
+              key="version"
+              title={t('ceph-storage-plugin~Version')}
+              isLoading={!subscriptionLoaded}
+              error={subscriptionLoaded && !ocsVersion}
+            >
+              {ocsVersion}
+              {ocsSubscriptionLoaded && packageManifestLoaded ? (
+                <Button type="button" isInline onClick={launchModal} variant="link">
+                  <BlueArrowCircleUpIcon className="co-icon-space-r" />
+                  {ocsSubscriptionLoaded ? ocsSubscription?.spec?.channel : null}
+                </Button>
+              ) : null}
+            </DetailItem>
+          ) : (
+            <DetailItem
+              key="version"
+              title={t('ceph-storage-plugin~Version')}
+              isLoading={!subscriptionLoaded}
+              error={subscriptionLoaded && !ocsVersion}
+            >
+              {ocsVersion}
```

Comment:
So, here we are displaying the ocsVersion. I don't think we should do again if no filters are available. @bipuladh